### PR TITLE
Update Search PHPDoc

### PR DESCRIPTION
**Pull request by cconard96**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -

Multiple class properties were undocumented including ones whose purpose is very difficult to understand just by their name and usage in the code. In fact, I still don't know what `SHORTSEP` and `LONGSEP` are specifically used for; just that they are used as separators in a GROUP_CONCAT.

Some other PHPDoc changes were made to improve type hinting of class strings and fixing incorrect information.

### CODE REVIEW

**Review by trasher — APPROVED** [review]

**Comment by orthagh** [comment]
LONGSEP is indeed for GROUP_CONCAT and separate several rows in one field
SHORTSEP is for simple CONCAT and permits to identify several colums in one field (most of the time id-name)

File: src/Search.php [hunk 1]
```diff
@@ -2344,10 +2420,14 @@ private static function isPossibleMetaSubitemOf(string $parent_itemtype, string
     }
 
     /**
-     * @since 0.85
+     * Gets the class to use if the specified itemtyoe extends one of the known reference types.
```
Comment (cedric-anne) [inline src/Search.php:2423]:
```suggestion
     * Gets the class to use if the specified itemtype extends one of the known reference types.
```

File: src/Search.php [hunk 2]
```diff
@@ -46,25 +46,101 @@
...
     const LBBR = '#LBBR#';
+
+    /**
+     * Placeholder for a <hr> line break
+     * @var string
+     */
     const LBHR = '#LBHR#';
 
+    /**
+     * Separator used for GROUP_CONCAT MySQL function.
```
Comment (cedric-anne) [inline src/Search.php:117]:
```suggestion
     * Separator used to separate values of a same element in CONCAT MySQL function.
```

File: src/Search.php [hunk 3]
```diff
@@ -46,25 +46,101 @@
...
+     *
+     * Unknown specific use cases.
+     * @var string
+     * @see LONGSEP
+     */
     const SHORTSEP = '$#$';
+
+    /**
+     * Separator used for GROUP_CONCAT MySQL function.
```
Comment (cedric-anne) [inline src/Search.php:126]:
```suggestion
     * Separator used to separate each element in GROUP_CONCAT MySQL function.
```

**Review by cedric-anne — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**